addons/account/models/account_account.py

```diff
@@ -478,44 +478,52 @@ def _set_opening_debit_credit(self, amount, field):
-                # If so, we need to merge all those lines into one before modifying its balance
-                opening_move_line = account_op_lines[0]
-                if len(account_op_lines) > 1:
-                    merge_write_cmd = [(1, opening_move_line.id, {'debit': op_aml_debit, 'credit': op_aml_credit, 'partner_id': None ,'name': _("Opening balance")})]
-                    unlink_write_cmd = [(2, line.id) for line in account_op_lines[1:]]
-                    opening_move.write({'line_ids': merge_write_cmd + unlink_write_cmd})
-
-                if amount:
-                    # modify the line
-                    opening_move_line.with_context(check_move_validity=False)[field] = amount
-                else:
-                    # delete the line (no need to keep a line with value = 0)
-                    opening_move_line.with_context(check_move_validity=False).unlink()
-
-            elif amount:
-                # create a new line, as none existed before
-                self.env['account.move.line'].with_context(check_move_validity=False).create({
-                        'name': _('Opening balance'),
-                        field: amount,
-                        'move_id': opening_move.id,
-                        'account_id': self.id,
-                })
-
-            # Then, we automatically balance the opening move, to make sure it stays valid
-            if not 'import_file' in self.env.context:
-                # When importing a file, avoid recomputing the opening move for each account and do it at the end, for better performances
-                self.company_id._auto_balance_opening_move()
+            with self.env['account.move']._check_balanced({'records': opening_move}):
+                # check whether we should create a new move line or modify an existing one
+                account_op_lines = self.env['account.move.line'].search([
+                    ('account_id', '=', self.id),
+                    ('move_id', '=', opening_move.id),
+                    (field, '!=', False),
+                    (field, '!=', 0.0),
+                ]) # 0.0 condition important for import
+
+                if account_op_lines:
+                    op_aml_debit = sum(account_op_lines.mapped('debit'))
+                    op_aml_credit = sum(account_op_lines.mapped('credit'))
+
+                    # There might be more than one line on this account if the opening entry was manually edited
+                    # If so, we need to merge all those lines into one before modifying its balance
+                    opening_move_line = account_op_lines[0]
+                    if len(account_op_lines) > 1:
+                        merge_write_cmd = [Command.update(opening_move_line.id, {
+                            'debit': op_aml_debit,
+                            'credit': op_aml_credit,
+                            'partner_id': None,
```

Comment:
Why `None` instead of `False` ? I see it was already there, but might as well clean it for better cohesion ?

Reply:
> There is still one useless `check_move_validity` in `test_unlink_timesheet`

Didn't want to do diff in the test :shrug: